monai/transforms/transforms.py

```diff
@@ -441,10 +444,514 @@ def __call__(self, img):
+            magnitude_range (2 ints): the random offsets on the grid will be generated from
+                `uniform[magnitude[0], magnitude[1])`.
+            prob (float): probability of returning a randomized affine grid.
+                defaults to 0.1, with 10% chance returns a randomized grid,
+                otherwise returns a `spatial_size` centered area centered extracted from the input image.
+            spatial_size (2 ints): specifying output image spatial size [h, w, d].
+            mode ('nearest'|'bilinear'): interpolation order. Defaults to 'bilinear'.
+            padding_mode ('zeros'|'border'|'reflection'): mode of handling out of range indices. Defaults to 'zeros'.
+            as_tensor_output (bool): the computation is implemented using pytorch tensors, this option specifies
+                whether to convert it back to numpy arrays.
+            device (torch.device): device on which the tensor will be allocated.
+
+        See also:
+            RandAffineGrid for the random affine paramters configurations.
+            Affine for the affine transformation parameters configurations.
+        """
+        self.rand_affine_grid = RandAffineGrid(rotate_range, shear_range, translate_range, scale_range, True, device)
+        self.resampler = Resample(padding_mode=padding_mode, as_tensor_output=as_tensor_output, device=device)
+
+        self.sigma_range = sigma_range
+        self.magnitude_range = magnitude_range
+        self.spatial_size = spatial_size
+        self.mode = mode
+        self.device = device
+
+        self.prob = prob
+        self.do_transform = False
+        self.rand_offset = None
+        self.magnitude = 1.0
+        self.sigma = 1.0
+
+    def set_random_state(self, seed=None, state=None):
+        self.rand_affine_grid.set_random_state(seed, state)
+        Randomizable.set_random_state(self, seed, state)
+        return self
+
+    def randomize(self, grid_size):
+        self.do_transform = self.R.rand() < self.prob
+        if self.do_transform:
+            self.rand_offset = self.R.uniform(-1., 1., [3] + list(grid_size))
+        self.magnitude = self.R.uniform(self.magnitude_range[0], self.magnitude_range[1])
+        self.sigma = self.R.uniform(self.sigma_range[0], self.sigma_range[1])
+
+    def __call__(self, img, spatial_size=None, mode=None):
+        """
+        Args:
+            img (ndarray or tensor): shape must be (num_channels, H, W, D),
+            spatial_size (2 ints): specifying output image spatial size [h, w, d].
+            mode ('nearest'|'bilinear'): interpolation order. Defaults to 'self.mode'.
+        """
+        spatial_size = spatial_size or self.spatial_size
+        mode = mode or self.mode
+        self.randomize(spatial_size)
+        grid = create_grid(spatial_size)
+        if self.do_transform:
+            grid = torch.tensor(grid).to(self.device)
+            gaussian = GaussianFilter(3, self.sigma, 3., device=self.device)
```

Comment:
As GaussianFilter contains "torch.nn.Parameter", should add "with torch.no_grad():" when run this operation?
Thanks.

Reply:
i set requires_grad to False https://github.com/Project-MONAI/MONAI/pull/131/files/871b074df3fada02a2d24507fe2a5ca81cddee8a?file-filters%5B%5D=.py#diff-94f123b1e6105a0be041ce6ef8461918R46

Reply:
OK, sounds good.